toolchain/check/convert.cpp

```diff
@@ -1213,11 +1213,43 @@ auto Convert(Context& context, SemIR::LocId loc_id, SemIR::InstId expr_id,
       CARBON_DIAGNOSTIC(ExplicitAsConversionFailure, Error,
                         "cannot convert value of type {0} to {1} with `as`",
                         TypeOfInstId, SemIR::TypeId);
-      return context.emitter().Build(loc_id,
-                                     target.kind == ConversionTarget::ExplicitAs
-                                         ? ExplicitAsConversionFailure
-                                         : ImplicitAsConversionFailure,
-                                     expr_id, target.type_id);
+      auto builder =
+          context.emitter().Build(loc_id,
```

Comment:
These notes end up making the diagnostic quite long. In any of these cases, could we replace the error with a different error instead of adding a note?